### ISSUE TITLE
[Eloquent] t265 node segfaults when tf_publish_rate != 0.0

### ISSUE THREAD

**Issue by Michael-Equi** [realsenseai/realsense-ros#1542]
When I run the t265 node with the following configuration I get the output below. I am also having trouble with setting `publish_odom_tf` to something other than false which I believe might be related to this issue. Note that I am running Foxy on  Ubuntu 20.04 so I am not sure if it is a distro specific issue or not since foxy is not officially supported yet. 

```
Configuration 
t265:
  camera:
    ros__parameters:
      device_type: 't265'
      serial_no: ''
      usb_port_id: ''

      odom_frame_id: 'odom'
      pose_frame_id: 't260_link'
      tf_publish_rate: 1.0
      publish_tf: true
      # publish_odom_tf: true

      enable_fisheye: true
      fisheye_height: 800
      fisheye_width: 848
      fisheye_fps: 30.0

      enable_accel: true
      enable_gyro: true

      publish_odom_tf: false

      unite_imu_method: 'linear_interpolation'

      tracking_module:
        enable_auto_exposure: true
        enable_dynamic_calibration: true
        enable_map_preservation: true
        enable_mapping: true
        enable_pose_jumping: false
        enable_relocalization: false
        frames_queue_size: 1

```

Node Output 

```
[INFO] [1606979273.967928471] [RealSenseCameraNode]: RealSense ROS v3.1.1
[INFO] [1606979273.967963647] [RealSenseCameraNode]: Built with LibRealSense v2.34.0
[INFO] [1606979273.967971112] [RealSenseCameraNode]: Running with LibRealSense v2.40.0
[WARN] [1606979273.967978062] [RealSenseCameraNode]: ***************************************************
[WARN] [1606979273.967985140] [RealSenseCameraNode]: ** running with a different librealsense version **
[WARN] [1606979273.967988860] [RealSenseCameraNode]: ** than the one the wrapper was compiled with!   **
[WARN] [1606979273.967992944] [RealSenseCameraNode]: ***************************************************
[INFO] [1606979273.976724443] [RealSenseCameraNode]: Device with serial number 11622110225 was found.

[INFO] [1606979273.976744517] [RealSenseCameraNode]: Device with physical ID 2-2-2 was found.
[INFO] [1606979273.976750178] [RealSenseCameraNode]: Device with name Intel RealSense T265 was found.
[INFO] [1606979273.976953356] [RealSenseCameraNode]: Device with port number 2-2 was found.
[INFO] [1606979273.978014862] [RealSenseCameraNode]: getParameters...
[INFO] [1606979273.978110110] [RealSenseCameraNode]: 662
[WARN] [1606979273.979330737] [RealSenseCameraNode]: 188has_parameter(infra_width)
[WARN] [1606979273.979432563] [RealSenseCameraNode]: 188has_parameter(infra_height)
[WARN] [1606979273.979528109] [RealSenseCameraNode]: 188has_parameter(infra_fps)
[WARN] [1606979273.979711430] [RealSenseCameraNode]: 188has_parameter(infra_width)
[WARN] [1606979273.979806757] [RealSenseCameraNode]: 188has_parameter(infra_height)
[WARN] [1606979273.979905241] [RealSenseCameraNode]: 188has_parameter(infra_fps)
[WARN] [1606979273.980772717] [RealSenseCameraNode]: 188has_parameter(fisheye_width)
[WARN] [1606979273.980873792] [RealSenseCameraNode]: 188has_parameter(fisheye_height)
[WARN] [1606979273.980975947] [RealSenseCameraNode]: 188has_parameter(fisheye_fps)
[WARN] [1606979273.981157321] [RealSenseCameraNode]: 188has_parameter(fisheye_width)
[WARN] [1606979273.981247752] [RealSenseCameraNode]: 188has_parameter(fisheye_height)
[WARN] [1606979273.981352007] [RealSenseCameraNode]: 188has_parameter(fisheye_fps)
[INFO] [1606979273.985025580] [RealSenseCameraNode]: setupDevice...
[INFO] [1606979273.985041587] [RealSenseCameraNode]: JSON file is not provided
[INFO] [1606979273.985052936] [RealSenseCameraNode]: Device Name: Intel RealSense T265
[INFO] [1606979273.985061833] [RealSenseCameraNode]: Device Serial No: 11622110225
[INFO] [1606979273.985067978] [RealSenseCameraNode]: Device physical port: 2-2-2
[INFO] [1606979273.985072693] [RealSenseCameraNode]: Device FW version: 0.2.0.951
[INFO] [1606979273.985077175] [RealSenseCameraNode]: Device Product ID: 0x0B37
[INFO] [1606979273.985081637] [RealSenseCameraNode]: Enable PointCloud: Off
[INFO] [1606979273.985086275] [RealSenseCameraNode]: Align Depth: Off
[INFO] [1606979273.985093455] [RealSenseCameraNode]: Sync Mode: Off
[INFO] [1606979273.985101942] [RealSenseCameraNode]: Device Sensors: 
[INFO] [1606979273.985122148] [RealSenseCameraNode]: Tracking Module was found.
[INFO] [1606979273.985131638] [RealSenseCameraNode]: (Depth, 0) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985137533] [RealSenseCameraNode]: (Color, 0) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985143966] [RealSenseCameraNode]: (Infrared, 0) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985150833] [RealSenseCameraNode]: (Infrared, 1) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985155679] [RealSenseCameraNode]: (Infrared, 2) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985160724] [RealSenseCameraNode]: (Fisheye, 0) sensor isn't supported by current device! -- Skipping...
[INFO] [1606979273.985167107] [RealSenseCameraNode]: num_filters: 0
[INFO] [1606979273.985172165] [RealSenseCameraNode]: Setting Dynamic reconfig parameters.
[INFO] [1606979273.986043807] [RealSenseCameraNode]: Done Setting Dynamic reconfig parameters.
[WARN] [1606979273.986104419] [RealSenseCameraNode]: Given stream configuration is not supported by the device!  Stream: Fisheye, Stream Index: 1, Width: 640, Height: 480, FPS: 30
[WARN] [1606979273.986121009] [RealSenseCameraNode]: Given stream configuration is not supported by the device!  Stream: Fisheye, Stream Index: 2, Width: 640, Height: 480, FPS: 30
[INFO] [1606979273.986140500] [RealSenseCameraNode]: setupPublishers...
[INFO] [1606979273.986146973] [RealSenseCameraNode]: Start publisher IMU
[INFO] [1606979273.987159349] [RealSenseCameraNode]: setupStreams...
[INFO] [1606979274.010121916] [RealSenseCameraNode]: SELECTED BASE:Pose, 0
[WARN] [1606979274.010240714] [RealSenseCameraNode]: Publishing dynamic camera transforms (/tf) at 1 Hz
[1]    60817 segmentation fault (core dumped)   --ros-args -r __ns:=/t265 --params-file
```

**Comment by doronhi** [comment]
Thanks for your report. This bug is common to eloquent as well as foxy and is not limited to T265.
You can now checkout the [foxy branch](https://github.com/IntelRealSense/realsense-ros/tree/foxy)
Fixed in #1586.

**Comment by RealSenseSupport** [comment]
Hi,

Will you be needing further help with this?

*If we don’t hear from you in 7 days, this issue will be closed.*

Thanks
